storage/interface.go

```diff
@@ -223,6 +223,17 @@ type SelectHints struct {
 	// When disabled, the result may contain samples outside the queried time range but Select() performances
 	// may be improved.
 	DisableTrimming bool
+
+	// Projection hints. They are currently unused in the Prometheus promql engine but can be used by different
```

Comment:
Should we give hints WHO is using it so if we need to make changes for Prometheus we know who we impact?

Or... what are the alternatives to this model? Can Thanos use context values? Should we add `Extra any` field for those cases?

Reply:
This should work for storages that are not called by Thanos PromQL engine too i think - I think it makes sense to have it upstream here! We of course should populaet them in the engine here too - but i would do that in followup. Having them here unlocks populating them in the Thanos engine and once that works great we can backport it here too!